schema/user.json

```diff
@@ -1,29 +1,59 @@
 {
+	"$schema": "https://json-schema.org/draft/2020-12/schema",
 	"$id": "https://wordpress.org/schema/user.json",
+
+	"title": "A site user.",
+	"description": "An individual site user, which can be referenced by other objects (for example, posts) as being the author/owner of that object.",
+
 	"type": "object",
+
 	"properties": {
 		"version": {
+			"description": "The schema version that this user object adheres to.",
 			"type": "integer",
 			"const": 1
 		},
 		"id": {
-			"type": "integer"
+			"description": "The user ID. It must be a positive integer.",
+			"type": "integer",
+			"minimum": 1
 		},
 		"login": {
+			"description": "The login or username for this user.",
 			"type": "string"
 		},
 		"name": {
-			"type": "string"
+			"description": "A display name for the user.\nWhen set as an object, it can be split up into the first name, last name, and preferred display name.\nWhen set as a string, that string will be treated as the display name.",
+			"type": [ "string", "object" ],
+			"properties": {
```

Comment:
Note, since this may appear a little odd: JSON Schema will ignore rules that don't apply to a particular type. `properties` only applies to the `object` type, so it will happily skip this bit when `name` is set to a `string`.